integreat_cms/cms/views/content_revision_view.py

```diff
@@ -0,0 +1,363 @@
+            content_name = "page"
+        elif kwargs.get("poi_id"):
+            content_type = POI
+            content_name = "poi"
+
+        region = request.region
+        content = (
+            content_type.objects.filter(region=region)
+            .filter(id=kwargs.get(f"{content_name}_id"))
+            .first()
+        )
+        if not content:
+            raise Http404("No Page matches the given region or id.")
+
+        language = region.get_language_or_404(
+            kwargs.get("language_slug"), only_active=True
+        )
+        content_translations = content.translations.filter(language=language)
+
+        if not content_translations.exists():
+            return redirect(
+                f"edit_{content_name}",
+                **{
+                    f"{content_name}_id": content.id,
+                    "region_slug": region.slug,
+                    "language_slug": language.slug,
+                },
+            )
+
+        if not self.can_change_content(request, content):
+            messages.warning(
+                request,
+                _("You don't have the permission to restore versions of this page."),
+            )
+
+        selected_revision = content_translations.filter(
+            version=kwargs.get("selected_revision", content_translations.count())
+        )
+
+        if not selected_revision.exists():
+            messages.error(request, _("This revision does not exist."))
+            return redirect(
+                f"{content_name}_revisions",
+                **{
+                    f"{content_name}_id": content.id,
+                    "region_slug": region.slug,
+                    "language_slug": language.slug,
+                },
+            )
+
+        if content.archived:
+            messages.warning(
+                request,
+                _("You cannot restore versions of this page because it is archived."),
+            )
+
+        kwargs = {
+            "region_slug": region.slug,
+            "language_slug": language.slug,
+            f"{content_name}_id": content.id,
+        }
+        revision = reverse(f"{content_name}_revisions", kwargs=kwargs)
+        form = reverse(f"edit_{content_name}", kwargs=kwargs)
+
+        return render(
+            request,
+            self.template_name,
+            {
+                **self.get_context(request, *args, **kwargs),
+                "content_name": content_name,
+                "content": content,
+                "content_translations": content_translations,
+                "api_revision": content_translations.filter(
+                    status=status.PUBLIC
+                ).first(),
+                "selected_revision": selected_revision.first(),
+                "language": language,
+                "form": form,
+                "revision": revision,
+            },
+        )
+
+    # pylint: disable=too-many-branches
+    def post(self, request, *args, **kwargs):
```

Comment:
There are quite a few duplications in `get()` and `post()`, whenever you notice this you can extract pieces of code into `dispatch()` which will be called in every case.